### ISSUE TITLE
Saved intermediary checkpoint's epoch is not honored when the checkpoint is loaded

### ISSUE THREAD

**Issue by 3dewberries** [ferdous-alam/GenCAD#23]
Quick story of what I wanted to achieve:

As the training process takes quite some time ( CSR and CCIP models training combined will take around a week to train on my PC ).
So I analyzed sources a bit and found the parameter "save_every" in the autoencoder configuration class ConfigAE.
By default this parameter is set to 50. I have set it to 1, to make the model training checkpoint to be saved after each epoch - I noticed it is done in the directory "results/test/CSR/trained_models" : 

<img width="1576" height="187" alt="Image" src="https://github.com/user-attachments/assets/73269fea-62b2-4ba7-aef3-315679180f78" />

So far so good. The goal was to be able to interrupt the process after any epoch and resume it at a later time.
But when I try to use the saved checkpoint file as a starting point, the saved epoch within it is not honored, and the training process starts again from epoch 1, even though the trainer script reports the CSR checkpoint is successfuly loaded:

<img width="2424" height="279" alt="Image" src="https://github.com/user-attachments/assets/378861c0-d19f-495c-8c9a-0194508426db" />

Also the optimizer's and scheduler's state_dict is saved but not loaded.

**Comment by 3dewberries** [comment]
Updated the autoencoder's trainer to support epoch, optimizer & scheduler state_dict when loading the checkpoint ( in the commit: https://github.com/3dewberries/GenCAD/commit/69de080a20fbd65bf0e4f4d9ee360510256c0994 )
